trio/_core/_run.py

```diff
@@ -2084,7 +2084,10 @@ def unrolled_run(runner, async_fn, args, host_uses_signal_set_wakeup_fd=False):
                 # seeded for tests.
                 batch.sort(key=lambda t: t._counter)
             runner.runq.clear()
-            _r.shuffle(batch)
+            # 50% chance of reversing the batch, this way each task
+            # can appear before/after any other task.
+            if _r.random() < 0.5:
+                batch = batch[::-1]
```

Comment:
I guess `batch.reverse()` can only be more efficient since it reverses the list in place?

Reply:
Yeah:

```python
In [1]: l = list(range(100))                                                              

In [2]: %timeit l[::-1]                                                                   
480 ns ± 2.71 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: %timeit l.reverse()                                                               
120 ns ± 5.19 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
Good point, I didn't remember it has a method to sort in-place.